### PR TITLE
Add longer accuracy test on CI

### PR DESCRIPTION
**Pull request by merrymercy**

### CODE REVIEW

File: test/srt/test_eval_accuracy_large.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    popen_launch_server,
+)
+
+
+class TestEvalAccuracyLarge(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        cls.model = DEFAULT_MODEL_NAME_FOR_TEST
+        cls.base_url = "http://127.0.0.1:9157"
```
Comment (zhyncs) [inline test/srt/test_eval_accuracy_large.py]: e2e, unit, accuracy should use different ports

File: test/srt/test_serving_throughput.py [hunk 2]
```diff
@@ -55,21 +56,30 @@ def run_test(self, disable_radix_cache, disable_flashinfer, chunked_prefill_size
...
 
     def test_default(self):
-        self.run_test(
+        res = self.run_test(
             disable_radix_cache=False,
             disable_flashinfer=False,
             chunked_prefill_size=-1,
         )
 
+        if os.getenv("CI", "false") == "true":
```
Comment (zhyncs) [inline test/srt/test_serving_throughput.py]: Why use CI?

File: test/srt/test_serving_throughput.py [hunk 3]
```diff
@@ -55,21 +56,30 @@ def run_test(self, disable_radix_cache, disable_flashinfer, chunked_prefill_size
...
+
     def test_default_without_radix_cache(self):
-        self.run_test(
+        res = self.run_test(
             disable_radix_cache=True,
             disable_flashinfer=False,
             chunked_prefill_size=-1,
         )
 
+        if os.getenv("CI", "false") == "true":
```
Comment (zhyncs) [inline test/srt/test_serving_throughput.py]: same

File: .github/workflows/accuracy-test.yml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    - name: Install dependencies
+      run: |
+        source $HOME/venv/bin/activate
+        echo "$HOME/venv/bin" >> $GITHUB_PATH
+
+        pip install --upgrade pip
+        pip install -e "python[all]"
+        pip install flashinfer -i https://flashinfer.ai/whl/cu121/torch2.4/ --force-reinstall
+
+        git clone https://github.com/merrymercy/human-eval.git
```
Comment (zhyncs) [inline .github/workflows/accuracy-test.yml:38]: Why not use https://github.com/openai/human-eval?

File: .github/workflows/accuracy-test.yml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        cd human-eval
+        pip install -e .
+
+    - name: Evaluate MMLU
+      run: |
+        cd test/srt
+        python3 -m unittest test_eval_accuracy_large.TestEvalAccuracyLarge.test_mmlu
+
+    - name: Evaluate HumanEval
```
Comment (zhyncs) [inline .github/workflows/accuracy-test.yml]: May we include gsm8k? It's sensitive to precision changes.

File: .github/workflows/accuracy-test.yml (hunk 4 above)
Comment (merrymercy) [reply .github/workflows/accuracy-test.yml:38]: The official one cannot run directly. We have to disable a warning https://github.com/merrymercy/human-eval/commit/94e08e6ede07e23554856e523048ac69be91059b

File: .github/workflows/accuracy-test.yml (hunk 5 above)
Comment (merrymercy) [reply .github/workflows/accuracy-test.yml]: This one is also very sensitive and it is the one that helps us find this bug https://github.com/sgl-project/sglang/pull/888. I have ported gqpa, math, mmlu, humaneval https://github.com/sgl-project/sglang/tree/main/python/sglang/test.
We can port https://github.com/openai/simple-evals/blob/main/mgsm_eval.py later.

Comment (zhyncs) [reply .github/workflows/accuracy-test.yml]: ok

Comment (merrymercy) [reply .github/workflows/accuracy-test.yml]: added gsm test cases

File: test/srt/test_serving_throughput.py [hunk 6]
```diff
@@ -55,21 +56,30 @@ def run_test(self, disable_radix_cache, disable_flashinfer, chunked_prefill_size
...
-        self.run_test(
+        res = self.run_test(
             disable_radix_cache=False,
             disable_flashinfer=False,
             chunked_prefill_size=-1,
         )
 
+        if os.getenv("SGLANG_IS_IN_CI", "false") == "true":
+            # A100 performance
+            assert res["output_token_throughput"] >= 1300
```
Comment (zhyncs) [inline test/srt/test_serving_throughput.py]: Why does it show a KeyError?

ref https://github.com/sgl-project/sglang/actions/runs/10348574527/job/28641104727

```
100%|██████████| 400/400 [04:40<00:00,  1.42it/s]
/usr/lib/python3.10/subprocess.py:1072: ResourceWarning: subprocess 3619578 is still running
  _warn("subprocess %s is still running" % self.pid,
ResourceWarning: Enable tracemalloc to get the object allocation traceback
E
======================================================================
ERROR: test_default (test_serving_throughput.TestServingThroughput)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/data/zhyncs/runner-v1/actions-runner/_work/sglang/sglang/test/srt/test_serving_throughput.py", line 70, in test_default
    assert res["output_token_throughput"] >= 1300
KeyError: 'output_token_throughput'
```

Comment (merrymercy) [reply test/srt/test_serving_throughput.py]: fixed.

**Review by zhyncs — APPROVED** [review]

**Comment by zhyncs** [comment]
nit: May we also change this in this PR? https://github.com/sgl-project/sglang/blob/89f23a5178769ce867e8fb3af3cb44da2f5399ec/python/sglang/srt/managers/tp_worker.py#L67-L68

**Comment by zhyncs** [comment]
> nit: May we also change this in this PR?
> 
> https://github.com/sgl-project/sglang/blob/89f23a5178769ce867e8fb3af3cb44da2f5399ec/python/sglang/srt/managers/tp_worker.py#L67-L68

ignore this one

**Pull request merged**